release/ray_release/test.py

```diff
@@ -83,6 +123,65 @@ def get_anyscale_byod_image(self) -> str:
         """
         return f"{DATAPLANE_ECR}/{self.get_byod_repo()}:{self.get_byod_image_tag()}"
 
+    def add_test_result(self, result: Result) -> None:
+        """
+        Add test result to test object
+        """
+        self.get_test_results().insert(0, TestResult.from_result(result))
+
+    def get_test_results(self, limit: int = 10) -> List[TestResult]:
+        """
+        Get test result from test object, or s3
+        """
+        if self.test_results is not None:
+            return self.test_results
+
+        s3_client = boto3.client("s3")
+        files = sorted(
+            s3_client.list_objects_v2(
+                Bucket=AWS_BUCKET,
+                Prefix=f"{AWS_TEST_RESULT_KEY}/{self.get_name()}-",
+            ).get("Contents", []),
+            key=lambda file: int(file["LastModified"].strftime("%s")),
+            reverse=True,
+        )[:limit]
+        self.test_results = [
+            TestResult.from_dict(
+                json.loads(
+                    s3_client.get_object(
+                        Bucket=AWS_BUCKET,
+                        Key=file["Key"],
+                    )
+                    .get("Body")
+                    .read()
+                    .decode("utf-8")
+                )
+            )
+            for file in files
+        ]
+        return self.test_results
+
+    def persist_result_to_s3(self, result: Result) -> bool:
```

Comment:
why are `add_test_result` and `persist_result_to_s3` two different functions?

or maybe I am still not very clear about the data model here :(

Reply:
good catch ;), let me just get rid of add_test_result to simplify the api